tests/migrations/test_autodetector.py

```diff
@@ -1304,6 +1307,33 @@ def test_different_regex_does_alter(self):
         self.assertNumberMigrations(changes, "testapp", 1)
         self.assertOperationTypes(changes, "testapp", 0, ["AlterField"])
 
+    def test_identical_callable_validator_doesnt_alter(self):
+        lt_now_validator = MinValueValidator(datetime.datetime.now)
+        from_state = ModelState(
+            "testapp", "model", [
+                ('dob', models.DateTimeField(validators=[MinValueValidator(datetime.datetime.now)])),
+            ]
+        )
+        to_state = ModelState(
+            "testapp", "model", [('dob', models.DateTimeField(validators=[lt_now_validator]))]
+        )
+        changes = self.get_changes([from_state], [to_state])
+        self.assertNumberMigrations(changes, "testapp", 0)
+
+    def test_different_callable_validator_does_alter(self):
+        lt_today_validator = MinValueValidator(datetime.datetime.today)
+        from_state = ModelState(
+            "testapp", "model", [
+                ('dob', models.DateTimeField(validators=[MinValueValidator(datetime.datetime.now)])),
+            ]
+        )
+        to_state = ModelState(
+            "testapp", "model", [('dob', models.DateTimeField(validators=[lt_today_validator]))]
+        )
+        changes = self.get_changes([from_state], [to_state])
+        self.assertNumberMigrations(changes, "testapp", 1)
+        self.assertOperationTypes(changes, "testapp", 0, ["AlterField"])
+
```

Comment:
I'm not sure there's much value in adding these two tests as they were really specific to the `SimpleLazyObject` issue. They pass with and without the changes to `BaseValidator.__call__`.

Reply:
I had the same thought -- just wasnt too sure. Either way, its gone